### PR TITLE
AEA-1258 Docs Review: Development Beginner

### PR DESCRIPTION
**Pull request by 5A11**
This PR updates the development-beginner part of the documentation.

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [x] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: docs/skill-guide.md [hunk 1]
```diff
@@ -930,9 +932,4 @@ We recommend you continue with the next step in the 'Getting Started' series:
 
 - <a href="../core-components-2">Core components (Part 2)</a>
 
-
-### Relevant deep-dives
```
Comment (DavidMinarsch) [inline docs/skill-guide.md:934]: Please keep section and link to the `generic buyer/seller` guide here

File: docs/quickstart.md [hunk 2]
```diff
@@ -199,11 +198,19 @@ aea add skill fetchai/echo:0.13.0
...
 </details>
 
-## Communication via envelopes and messages
+### Echo skill
 
-AEAs use envelopes containing messages for communication. To learn more check out the <a href="../core-components-1/">next section</a> in the getting started series.
+Just like humans, AEAs can have _skills_ to achieve their tasks. As an agent developer, you can create skills to add to your own AEAs. You can also choose to publish your skills so others add them to their AEAs. More details on skills can be found on <a href="../skill/"> this page </a>.
 
-## Usage of the stub connection
+The above agent has an <a href="https://aea-registry.fetch.ai/details/skill/fetchai/echo/latest" target="_blank">echo skill</a> which simply echoes any messages it receives back to its sender.
```
Comment (5A11) [inline docs/quickstart.md:205]: Here is the second part

File: docs/quickstart.md [hunk 3]
```diff
@@ -169,9 +168,9 @@ AEA configurations successfully initialized: {'author': 'fetchai'}
 
 ## Echo skill demo
 
-The echo skill demo is a simple demo that introduces you to the main business logic components of an AEA. The <a href="https://aea-registry.fetch.ai/details/skill/fetchai/echo/latest" target="_blank">echo skill</a> simply echoes received messages back to the sender.
```
[the remark this answers is not in the document]
Comment (5A11) [reply docs/quickstart.md:172]: The this sentence is still there. Just broken up to two parts. The second part is below.

File: docs/quickstart.md (hunk 2 above)
Comment (5A11) [reply docs/quickstart.md:205]: Added link to registry.

File: docs/skill-guide.md [hunk 4]
```diff
@@ -116,6 +116,11 @@ Searches are proactive and, as such, well placed in a <a href="../api/skills/bas
 
 We place this code in `my_aea/skills/my_search/behaviours.py`. Ensure you replace the `fetchai` author in this line `from packages.fetchai.skills.my_search.dialogues import OefSearchDialogues` with your author handle (run `aea init` to set or check the author name).
 
+<div class="admonition note">
+  <p class="admonition-title">Note</p>
+  <p> Note that the import paths to agent packages, fo example `packages.fetchai.skills.my_search.dialogues` above, are not actual paths. Package files always reside in your AEA's folder, either under a specific package directory (e.g. connection, protocol, skill) if the package is custom built, or under `vendor` if it is pulled from the registry. These paths are virtual and created automatically when an AEA is run. </p>
+</div>
```
Comment (DavidMinarsch) [inline docs/skill-guide.md:122]: Point to https://docs.fetch.ai/aea/package-imports/ for more details.

File: mkdocs.yml [hunk 5]
```diff
@@ -44,7 +44,7 @@ nav:
     - AEA and web frameworks: 'aea-vs-mvc.md'
     - Build your first skill - search & discovery: 'skill-guide.md'
     - Core components - Part 2: 'core-components-2.md'
-    - How AEAs talk to each other - interaction protocols: 'interaction-protocol.md'
```
Comment (DavidMinarsch) [inline mkdocs.yml:47]: Please revert

Comment (5A11) [reply mkdocs.yml]: I am not really a fan of long sentences in titles, but reverted.

File: docs/generic-skills-step-by-step.md [hunk 6]
```diff
@@ -402,7 +400,7 @@ Below the unused `teardown` function, we continue by adding the following code:
 
 The above code handles an unidentified dialogue by responding to the sender with a `DefaultMessage` containing the appropriate error information.
 
-The next code block handles the `CFP` message, paste the code below the `_handle_unidentified_dialogue` function:
+The next code block handles `CFP` (call-for-protposal) negotiation messages. Paste the following code below the `_handle_unidentified_dialogue` function:
```
Comment (DavidMinarsch) [inline docs/generic-skills-step-by-step.md:404]:
```suggestion
The next code block handles `CFP` (call-for-proposal) negotiation messages. Paste the following code below the `_handle_unidentified_dialogue` function:

```

File: docs/generic-skills-step-by-step.md [hunk 7]
```diff
@@ -958,9 +956,9 @@ class GenericStrategy(Model):
         self._data_for_sale = data_for_sale
 ```
 
-We initialise the strategy class. We are trying to read the strategy variables from the YAML file. If this is not possible we specified some default values.
+In the above code snippet, we initialise the strategy class by trying to read the variables specific to the strategy from a YAML configuration file. If any variable isn't provided, some default values will be used.
```
Comment (DavidMinarsch) [inline docs/generic-skills-step-by-step.md:959]:
```suggestion
In the above code snippet, we initialise the strategy class by trying to read the variables specific to the strategy from a YAML configuration file. If any variable is not provided, some default values will be used.
```

File: docs/generic-skills-step-by-step.md [hunk 8]
```diff
@@ -3050,13 +3045,11 @@ class SigningDialogues(Model, BaseSigningDialogues):
         )
 ```
 
-The dialogues class stores dialogue with each AEA and other AEA components so we can have access to previous messages and enable us to identify possible communications problems between the `my_generic_seller` AEA and the `my_generic_buyer` AEA.
+The various dialogues classes in the above code snippet store dialogues with other AEAs, services and components, (e.g. SOEF search node, Ledgers, Decision maker). They expose useful methods to manipulate these interactions, access previous messages, and enable us to identify possible communications problems between `my_generic_seller` and `my_generic_buyer` AEAs.
```
Comment (DavidMinarsch) [inline docs/generic-skills-step-by-step.md:3048]:
```suggestion
The various dialogues classes in the above code snippet store dialogues with other AEAs, services and components, (e.g. SOEF search node via the `fetchai/soef` connection, ledgers via the `fetchai/ledger` connection, the decision maker). They expose useful methods to manipulate these interactions, access previous messages, and enable us to identify possible communications problems between `my_generic_seller` and `my_generic_buyer` AEAs.
```

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**